src/Streamly/FileSystem/Handle.hs

```diff
@@ -9,68 +9,76 @@
+-- 'Streamly.Prelude.before' combinator may be used to do that on a streaming
+-- combinator.  To restrict the length of read or write use the stream trimming
+-- operations like 'Streamly.Prelude.take'.
+--
+-- Note that a 'Handle' is inherently stateful, therefore, we cannot use these
+-- APIs from multiple threads without serialization; reading or writing in one
+-- thread would affect the file position for other threads.
 --
 -- For additional, experimental APIs take a look at
 -- "Streamly.Internal.FileSystem.Handle" module.
+
+-- Design notes:
 --
--- = Performance Notes
+-- By design, file handle IO APIs are quite similar to
+-- "Streamly.Data.Array.Foreign" read write APIs. In that regard, arrays can be
+-- considered as in-memory files or files can be considered as on-disk arrays.
 --
--- In some cases the stream type based APIs in the
--- "Streamly.Internal.FileSystem.Handle" module may be more efficient compared
--- to the unfold/fold based APIs exposed from this module because of better
--- fusion by GHC. However, with the streamly fusion GHC plugin (upcoming) these
--- APIs would perform as well as the stream based APIs in all cases.
-
--- IO APIs are divided into two categories, sequential streaming IO APIs and
--- random access IO APIs.
-
 module Streamly.FileSystem.Handle
     (
-    -- * Sequential/Streaming IO
-    -- | Stream data to or from a file or device sequentially.  When reading,
-    -- the stream is lazy and generated on-demand as the consumer consumes it.
-    -- Read IO requests to the IO device are performed in chunks limited to a
-    -- maximum size of 32KiB, this is referred to as
-    -- 'Streamly.Internal.Data.Array.Foreign.Type.defaultChunkSize' in the
-    -- documentation. One IO request may or may not read the full
-    -- chunk. If the whole stream is not consumed, it is possible that we may
-    -- read slightly more from the IO device than what the consumer needed.
-    -- Unless specified otherwise in the API, writes are collected into chunks
-    -- of 'Streamly.Internal.Data.Array.Foreign.Type.defaultChunkSize' before they
-    -- are written to the IO device.
+    -- * Singleton IO
+    -- | Read or write a single buffer.
+      getChunk
+    , putChunk
+
+    -- * Streaming IO
+    -- | Read or write a stream of data to or from a file or device
+    -- sequentially.  Read requests to the IO device are performed in chunks
+    -- limited to a maximum size of
+    -- 'Streamly.Internal.System.IO.defaultChunkSize'.  Note that the size of
+    -- the actual chunks in the resulting stream may be less than the
+    -- @defaultChunkSize@ but it can never exceed it.  If the whole stream is
+    -- not consumed, it is possible that we may have read slightly more from
+    -- the IO device than what the consumer needed.  Unless specified otherwise
+    -- in the API, writes are collected into chunks of
+    -- 'Streamly.Internal.System.IO.defaultChunkSize' before they are written
+    -- to the IO device.
```

Comment:
```suggestion
    -- * Streaming IO
    -- | Read or write a stream of data to or from a file or device
    -- sequentially.  
    --
    -- Read requests to the IO device are performed in chunks
    -- limited to a maximum size of
    -- 'Streamly.Internal.System.IO.defaultChunkSize'.  Note that the size of
    -- the actual chunks in the resulting stream may be less than the
    -- @defaultChunkSize@ but it can never exceed it.  If the whole stream is
    -- not consumed, it is possible that we may have read slightly more from
    -- the IO device than what the consumer needed.  
    --
    -- Writes are collected into chunks of
    -- 'Streamly.Internal.System.IO.defaultChunkSize' before they are written
    -- to the IO device unless specified otherwise in the API
```

Reply:
Basically split them a little. When combined, suddenly you read about `write` API with the whole context of `read` in mind.

Reply:
I knew about it, but then I thought maybe its fine. will change.